app/routes/endemics/date-of-visit.js

```diff
@@ -179,33 +211,12 @@ module.exports = [
         const formattedTypeOfLivestock = (isPigs || isSheep) ? typeOfLivestock : `${typeOfLivestock} cattle`
         const dateOfVisit = new Date(request.payload[labels.year], request.payload[labels.month] - 1, request.payload[labels.day])
         const { isValid, reason } = isValidDateOfVisit(dateOfVisit, typeOfReview, previousClaims, latestVetVisitApplication)
-        const mainMessage = { url: '#' }
+        let mainMessage = { url: '#' }
         let backToPageMessage = 'Enter the date the vet last visited your farm for this review.'
         if (!isValid) {
-          switch (reason) {
-            case dateOfVetVisitExceptions.reviewWithin10:
-              mainMessage.text = 'There must be at least 10 months between your reviews.'
-              mainMessage.url = 'https://www.gov.uk/guidance/farmers-how-to-apply-for-funding-to-improve-animal-health-and-welfare#timing-of-reviews-and-follow-ups'
-              break
-            case dateOfVetVisitExceptions.rejectedReview:
-              mainMessage.text = `${organisation?.name} - SBI ${organisation?.sbi} had a failed review claim for ${formattedTypeOfLivestock} in the last 10 months.`
-              break
-            case dateOfVetVisitExceptions.noReview:
-              mainMessage.text = 'There must be no more than 10 months between your reviews and follow-ups.'
-              mainMessage.url = 'https://www.gov.uk/guidance/farmers-how-to-apply-for-funding-to-improve-animal-health-and-welfare#timing-of-reviews-and-follow-ups'
-              backToPageMessage = 'Enter the date the vet last visited your farm for this follow-up.'
-              break
-            case dateOfVetVisitExceptions.endemicsWithin10:
-              mainMessage.text = 'There must be at least 10 months between your follow-ups.'
-              mainMessage.url = 'https://www.gov.uk/guidance/farmers-how-to-apply-for-funding-to-improve-animal-health-and-welfare#timing-of-reviews-and-follow-ups'
-              backToPageMessage = 'Enter the date the vet last visited your farm for this follow-up.'
-              break
-            case dateOfVetVisitExceptions.claimEndemicsBeforeReviewPayment:
-              mainMessage.text = 'Your review claim must have been approved before you claim for the follow-up that happened after it.'
-              mainMessage.url = 'https://www.gov.uk/guidance/farmers-how-to-apply-for-funding-to-improve-animal-health-and-welfare#timing-of-reviews-and-follow-ups'
-              backToPageMessage = 'Enter the date the vet last visited your farm for this follow-up'
-              break
-          }
+          const messagesForInvalidOption = getMainMessage(reason, dateOfVetVisitExceptions, backToPageMessage, organisation, formattedTypeOfLivestock)
+          mainMessage = messagesForInvalidOption.mainMessage
+          backToPageMessage = messagesForInvalidOption.backToPageMessage
```

Comment:
Can the function return mainMessage and backToPageMessage as destructured elements to save creating a let variable for main message?

Would you also be able to rename it to getMessage as it not only gets the main message but also the back to page message

Reply:
we could put it back when need it , or  just drop a note to Dave that this is the case . and I will revert that change